Extension/src/LanguageServer/client.ts

```diff
@@ -3504,12 +3507,32 @@ export class DefaultClient implements Client {
                     character: range.end.character,
                     line: range.end.line
                 }
-            }
+            },
+            copyToClipboard: false
         };
 
+        if (copy) {
+            params.copyToClipboard = true;
+        }
+
         const result: CreateDeclarationOrDefinitionResult = await this.languageClient.sendRequest(CreateDeclarationOrDefinitionRequest, params);
+
+        if (result.edit === undefined) {
+            return;
+        }
         // TODO: return specific errors info in result.
-        if (result.changes === undefined) {
+        if (result.edit.changes.length === undefined && result.clipboardText) {
+
+            if (!params.copyToClipboard) {
+                util.promptCDDFailed();
```

Comment:
Is this still necessary if the language server is already sending a message that CD failed when the command is not "copy"?

Reply:
The conditional `if (result.edit.changes.length === undefined && result.clipboardText)` is true either case "create..." or "copy....". If this logic should only be executed for "copy..." case then there should be a check in the conditional for that operation only.

Reply:
To clarify, what are these errors `util.promptCDDFailed()` and `util.promptCDDFailed(true)` meant for?

Reply:
Initially it was to notify the user visually if either CDD failed or CopytoClipboard failed. I think inititially we agreed to do both in our first spec meeting but now it makes sense not to show this notification for if CDD fails. I still think this notification is appropriate for if copyToClipboard fails since the user will have no other visual cue (overall this scenario will be extremely rare anyways).